### PR TITLE
Introduce `GetSubscriptionUpcomingInvoice` endpoint

### PR DESCRIPTION
**Pull request by vykhrystiuk**
## Summary
Introduce preview upcoming invoice endpoint `GetSubscriptionUpcomingInvoice` `GET /subscriptions/{id}/upcoming-invoice`

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/paths/subscriptions@{id}@upcoming-invoice.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  tags:
+    - Orders
+  summary: Retrieve subscription order's upcoming invoice
+  operationId: GetSubscriptionUpcomingInvoice
+  x-sdk-operation-name: getUpcomingInvoice
+  description: |-
+    Retrieve an upcoming invoice from the specified subscription order.
+
+    The endpoint is temporary before upcoming invoices get a complete
+    integration.
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@upcoming-invoice.yaml:15]: nitpick: suggestion based on [writing style guidance for operation descriptions](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#operations)

I also added an updated note about the endpoint status
```suggestion
    Retrieves an upcoming invoice for a specified subscription order.

    > **Note:** This endpoint is temporary until an integration
    is completed for upcoming invoices.
```

File: openapi/paths/subscriptions@{id}@upcoming-invoice.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  parameters:
+    - $ref: ../components/parameters/collectionExpand.yaml
+  deprecated: true
+  responses:
+    '200':
+      description: Upcoming invoice is retrieved.
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@upcoming-invoice.yaml:21]: nitpick: the word 'is' is not needed here
```suggestion
      description: Upcoming invoice retrieved.
```

File: openapi/paths/subscriptions@{id}@upcoming-invoice.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+parameters:
+  - $ref: ../components/parameters/resourceId.yaml
+get:
+  x-products:
+    - Core
+  tags:
+    - Orders
+  summary: Retrieve subscription order's upcoming invoice
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@upcoming-invoice.yaml:8]: nitpick: where possible, we should avoid using possessives `'s`

I think we can also remove `subscription order` (this is described in the description)
```suggestion
  summary: Retrieve an upcoming invoice
```

**Review by noon-dawg — COMMENTED** [review]
Minor suggestions

**Comment by vykhrystiuk** [comment]
Thanks, @noon-dawg!

**Comment by noon-dawg** [comment]
LGTM!

File: openapi/paths/subscriptions@{id}@upcoming-invoice.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  operationId: GetSubscriptionUpcomingInvoice
+  x-sdk-operation-name: getUpcomingInvoice
+  description: |-
+    Retrieve an upcoming invoice from the specified subscription order.
+
+    The endpoint is temporary before upcoming invoices get a complete
+    integration.
+  parameters:
+    - $ref: ../components/parameters/collectionExpand.yaml
+  deprecated: true
```
Comment (2e3s) [inline openapi/paths/subscriptions@{id}@upcoming-invoice.yaml:18]: Already deprecated?
```suggestion
```

File: openapi/paths/subscriptions@{id}@upcoming-invoice.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  tags:
+    - Orders
+  summary: Retrieve an upcoming invoice
+  operationId: GetSubscriptionUpcomingInvoice
+  x-sdk-operation-name: getUpcomingInvoice
+  description: |-
+    Retrieves an upcoming invoice for a specified subscription order.
+
+    > **Note:** This endpoint is temporary until an integration
+    is completed for upcoming invoices.
```
[the remark this answers is not in the document]
Comment (noon-dawg) [reply openapi/paths/subscriptions@{id}@upcoming-invoice.yaml:15]: I reworded this a bit, but to be honest I'm also confused by it. I'm assuming that it is a temporary endpoint that will be replaced by a future implementation, which seems odd.

I've come across this message in: https://github.com/Rebilly/api-definitions/blob/f1886964d3d2cba361e4867b0977a8a69d93ae42/openapi/paths/subscriptions%40%7Bid%7D%40upcoming-invoices.yaml#L14-L15

**Review by colinbird — APPROVED** [review]

**Review by ninjasimon — APPROVED** [review]

**Review by 2e3s — APPROVED** [review]

**Pull request merged**